### PR TITLE
Replace our proxy script by squid proxy

### PR DESCRIPTION
**Pull request by jkonecny12**
Fix tests using the old proxy:
**proxy-kickstart**
**proxy-cmdline**
**proxy-auth**

### CODE REVIEW

File: functions.sh [hunk 1]
```diff
@@ -116,6 +116,43 @@ start_httpd() {
...
+
+    if [ -n "$2" ]; then
+        config="$2"
+    fi
+
+    # Starts a proxy server rooted in $proxy_root. The PID of the server will be
+    # written to $tmpdir/proxy-pid, and the URL for the server will be set in
+    # $proxy_url
+
+    # Tinyproxy must have rights for its folder
```
Comment (rvykydal) [inline functions.sh]: "Tinyproxy"

File: functions.sh [hunk 2]
```diff
@@ -116,6 +116,43 @@ start_httpd() {
...
+    chmod 777 ${proxy_root}
+
+    local scriptdir=${PWD}/scripts
+
+    # Copy configuration file for squid from confs folder to proxy_root
+    # and launch the proxy.
+    cp $scriptdir/confs/$config $proxy_root/squid.conf
+    local proxy_info="$(${scriptdir}/launch_proxy.sh "${proxy_root}")"
+
+    # Parse out the port and load PID
```
Comment (rvykydal) [inline functions.sh]: load PID?

File: proxy-cmdline.sh [hunk 3]
```diff
@@ -16,24 +16,47 @@
...
+    echo vnc inst.proxy=${proxy_url%%/*} inst.debug
 }
 
 prepare() {
     ks=$1
     tmpdir=$2
 
-    # Flatten the kickstart to include the proxy %pre script
-    ( cd "$(dirname ${ks})" && ksflatten -o ${tmpdir}/kickstart.ks -c "$(basename $ks)" )
+    scriptdir=$PWD/scripts
```
Comment (rvykydal) [inline proxy-cmdline.sh]: this scriptdir variable is not used?

File: functions.sh (hunk 1 above)
Comment (jkonecny12) [inline functions.sh]: Fixed

File: functions.sh (hunk 2 above)
Comment (jkonecny12) [inline functions.sh]: This was leftover from start_http copy. Fixed.

File: proxy-cmdline.sh (hunk 3 above)
Comment (jkonecny12) [inline proxy-cmdline.sh]: True thanks.

**Comment by jkonecny12** [comment]
**UPDATED:** Should be fine now.

**Pull request merged**